lib/stimulus_reflex/utils/sanity_checker.rb

```diff
@@ -29,6 +30,10 @@ def called_by_installer?
     def called_by_generate_config?
       ARGV.include? "stimulus_reflex:initializer"
     end
+
+    def called_by_rake?
+      caller.find { |c| c.include?("/gems/rake-") }
+    end
```

Comment:
Hmmm. After checking a third time I'm not sure if this catches all the possibilities.

If you upgrade an app from 3.4 to 3.5.0.pre0 and try to run it, it obviously fails with the expected error message:
```
StimulusReflex strongly suggests that you set default_url_options in your environment files. [...]
```

It also suggested that you can generate an initializer to change the config.

If you then try to run the initializer generator it still fails:
```bash
> rails generate stimulus_reflex:initializer
Heads up! 🔥

StimulusReflex strongly suggests that you set default_url_options in your environment files [...]
```

Reply:
Is there something more dramatic that we can do, like the rake equivalent of a mutex? I'm even open to temp files if it gets the job done.

Either way, I actively need help getting this over the finish line - especially because what I'm "supposed" to be doing is writing documentation. (How's that thesis coming along? 😉 )

Reply:
Based on a brief perusal of how to deal with mutexes, the best answer for long running Rake tasks is to generate a `lockfile` and then check if that lockfile exists, if not, then create a lockfile and do your business.

https://stackoverflow.com/questions/7321083/rake-check-if-already-running/7851642